pkg/controller/registrysyncer/registrysyncer.go

```diff
@@ -0,0 +1,336 @@
+		return fmt.Errorf("failed to construct controller: %w", err)
+	}
+
+	for _, m := range managers {
+		if err := c.Watch(
+			source.NewKindWithCache(&imagev1.ImageStream{}, m.GetCache()),
+			handlerFactory(testInputImageStreamTagFilterFactory()),
+		); err != nil {
+			return fmt.Errorf("failed to create watch for ImageStreams: %w", err)
+		}
+	}
+
+	r.log.Info("Successfully added reconciler to manager")
+	return nil
+}
+
+type objectFilter func(types.NamespacedName) bool
+
+// handlerFactory produces a handler that:
+// * Watches ImageStreams because ImageStreamTags do not support the watch verb
+// * Extracts all ImageStramTags out of the Image
+// * Filters out the ones that are not in use
+// Note: We can not use a predicate because that is directly applied on the source and the source yields ImageStreams, not ImageStreamTags
+// * Creates a reconcile.Request per cluster and ImageStreamTag
+func handlerFactory(filter objectFilter) handler.EventHandler {
+	return imagestreamtagmapper.New(func(in reconcile.Request) []reconcile.Request {
+		if !filter(in.NamespacedName) {
+			return nil
+		}
+		return []reconcile.Request{in}
+	})
+}
+
+type reconciler struct {
+	ctx              context.Context
+	log              *logrus.Entry
+	registryClients  map[string]ctrlruntimeclient.Client
+	pullSecretGetter func() []byte
+}
+
+func (r *reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error) {
+	log := r.log.WithField("request", req.String())
+	log.Info("Starting reconciliation")
+	err := r.reconcile(req, log)
+	if err != nil {
+		log.WithError(err).Error("Reconciliation failed")
+	} else {
+		log.Info("Finished reconciliation")
+	}
+	return reconcile.Result{}, controllerutil.SwallowIfTerminal(err)
+}
+
+func (r *reconciler) reconcile(req reconcile.Request, log *logrus.Entry) error {
+
+	isTags := map[string]*imagev1.ImageStreamTag{}
+	for clusterName, client := range r.registryClients {
+		imageStreamTag := &imagev1.ImageStreamTag{}
+		if err := client.Get(r.ctx, req.NamespacedName, imageStreamTag); err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Debugf("Source imageStreamTag %s not found on cluster %s", req.NamespacedName.String(), clusterName)
+				return nil
+			}
+			return fmt.Errorf("failed to get imageStreamTag %s from registry cluster: %w", req.NamespacedName.String(), err)
+		}
+		isTags[clusterName] = imageStreamTag
+	}
+
+	srcClusterName := findNewest(isTags)
+	if srcClusterName == "" {
+		// the isTag does NOT exist on both clusters
+		// This case is not an error but expected to happen occasionally
+		return nil
+	}
+	sourceImageStreamTag := isTags[srcClusterName]
+
+	imageStreamNameAndTag := strings.Split(req.Name, ":")
+	if n := len(imageStreamNameAndTag); n != 2 {
+		return fmt.Errorf("when splitting imagestreamtagname %s by : expected two results, got %d", req.Name, n)
+	}
+	imageStreamName, imageTag := imageStreamNameAndTag[0], imageStreamNameAndTag[1]
+	sourceImageStream := &imagev1.ImageStream{
```

Comment:
See the other comment, we have to GET this to make sure we correctly copy its settings over